### PR TITLE
Onramp functionality: button + hook

### PR DESCRIPTION
**Pull request by zeppelinnn**
Summary of changes:
- Mark a pesky auto-created dir to be ignored via .gitignore
- changed lint-staged to only run on push (matches main app behavior)
- added core button component, shared by sign-in and onramp
- added `useOnramp` hook
- added `<BuyCrypto />` button, which uses `useOnramp` under the hood
  - support light/dark theming

still TODO (in followup):
- onSuccess / onError callbacks
- integrate clientId to actually be used via /onramp main app route
- customizable variant for `contained` vs `outlined` button

Test plan:
- Published alpha version and used it in the `react-sdk-demo` repo by adding both the <BuyCrypto> button and a barebones html button to demonstrate the usage. Verified that when clicked, the onramp experience is as expected

### CODE REVIEW

**Comment by zeppelinnn** [comment]
Current dependencies on/for this PR:
* latest
  * **PR #121** <a href="https://app.graphite.dev/github/pr/fractalwagmi/react-sdk/121" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/fractalwagmi/react-sdk/121?utm_source=stack-comment).

**Review by dborstelmann — COMMENTED** [review]
kan should probably review

File: package.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "@fractalwagmi/react-sdk",
-  "version": "1.1.5",
+  "version": "1.1.6-alpha.10",
```
Comment (obber) [inline package.json:3]: Do you intend on committing this alpha version to the repo? I think you can deploy this alpha version by changing the `version` tag temporarily, pushing to npm, and then choose not to commit it, if you want.

File: package.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
   "description": "React SDK for signing in with Fractal",
   "main": "dist/cjs/index.js",
   "module": "dist/esm/index.js",
-  "types": "dist/esm/index.d.ts",
+"types": "dist/esm/index.d.ts",
```
Comment (obber) [inline package.json:7]: spacing is off

File: src/hooks/public/use-onramp.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+const POPUP_HEIGHT_PX = 672;
+
+interface UseOnrampOptions {
+  theme?: ButtonProps['theme'];
+}
+
+export const useOnramp = (options?: UseOnrampOptions) => {
+  const { theme } = options ?? { theme: 'light' };
+  const { clientId, user } = useContext(FractalSDKContext);
+  const { open: openPopup } = usePopupConnection({
```
Comment (obber) [inline src/hooks/public/use-onramp.ts:16]: Do you have logic that closes the popup when the onramp popup window is rejected by the user or the popup is closed without approval?

I think for those, we need to listen to the `connection` object. You can see a more involved example of this in [use-sign-transaction](https://github.com/fractalwagmi/react-sdk/blob/latest/src/hooks/public/use-sign-transaction.tsx#L105-L111).

I think `POPUP_CLOSED` and `ONRAMP_REJECTED` (the one you added) would be good to handle.

Comment (obber) [reply src/hooks/public/use-onramp.ts:16]: `usePopupConnection` returns both `connection` and a `close` method to facilitate this.

File: src/hooks/public/use-onramp.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    enabled: !user,
+    heightPx: POPUP_HEIGHT_PX,
+    platform: Platform.REACT_SDK,
+  });
+
+  const openOnrampWindow = useCallback(() => {
+    openPopup(`${ONRAMP_URL}?clientId=${clientId}&theme=${theme}`);
+  }, []);
```
Comment (obber) [inline src/hooks/public/use-onramp.ts:24]: If you want to promisify this callback, then `use-sign-transaction` is a good example of this paradigm as the `signTransaction` callback that it returns will resolve or reject with the appropriate SDKError object, making it awaitable.

**Comment by obber** [comment]
Also, do you want to add docs to the README while you're at it?

File: package.json (hunk 1 above)
Comment (zeppelinnn) [reply package.json:3]: good call i'll undo this for the moment

File: src/hooks/public/use-onramp.ts (hunk 3 above)
Comment (zeppelinnn) [reply src/hooks/public/use-onramp.ts:16]: yeah thats the plan. i was going to implement it in a separate PR but still keep the feature unreleased until that's all done

**Comment by zeppelinnn** [comment]
> Also, do you want to add docs to the README while you're at it?

yeah ill do this before releasing the functionality, there's several other things i still need to implement which i would want to document as well

File: src/hooks/public/use-onramp.ts (hunk 4 above)
Comment (zeppelinnn) [reply src/hooks/public/use-onramp.ts:24]: thanks, i'll take a look at that pattern. so if im understanding the advantage to promisifying is that we expose a way for the caller to get success/error state more easily within the normal flow control of opening the popup here?

Comment (obber) [reply src/hooks/public/use-onramp.ts:24]: Yeah, so the caller could do something like

```
try {
  await openOnrampWindow();
} catch (err) {
  if (err instanceof SomeOnrampErrorClass) {
    // do this.
  }
  // other handling
}
```

I think with something like opening the onramp window, I think it'll make sense to return a promise.

The argument for not returning a promise is if the result should be obviously consumed as part of the react component lifecycle, like a data fetcher hook, which I don't think this function really fits into that well.

The other argument would be if it's not returning as part of a singular event (like a stream of data or something), but I think in this instance, the function is either going to succeed or fail, if I understand correctly.

File: src/hooks/public/use-onramp.ts (hunk 3 above)
Comment (obber) [reply src/hooks/public/use-onramp.ts:16]: Ok, feel free to split it up how you see fit.

**Review by obber — APPROVED** [review]

**Comment by zeppelinnn** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: bwALICUdkFh7rikdIsxU.
See details on [graphite.dev](https://app.graphite.dev/github/pr/fractalwagmi/react-sdk/121)

**Pull request merged**